Demo/RCHSearchViewController.swift

```diff
@@ -52,94 +78,109 @@ class RCHSearchViewController: UIViewController, UISearchBarDelegate, UICollecti
-                print("Recommended Products Error")
-                return
-            }
-            self.productArray = productArray
             
-            if self.productArray.count == 0 {
-                self.showNoResults()
-            } else {
-                self.searchResultsCollectionView.isHidden = false
-                self.searchResultsCollectionView?.reloadData()
+            self.products = searchResult.products!
 
-            }
         }) { (responseObject, error) in
             print(error)
         }
-//        resetSearch()
     }
     
-    func resetSearch() {
-        view.endEditing(true)
-//        autocompleteArray = []
-        autocompleteTableView.reloadData()
-        autocompleteTableView.isHidden = true
-        searchProductsView.isHidden = false
+    func autocomplete(withQuery text: String) {
+        let autocompleteBuilder: RCHAutocompleteBuilder = RCHSDK.builderForAutocomplete(withQuery: text)
+        
+        RCHSDK.defaultClient().sendRequest(autocompleteBuilder.build(), success: { (responseObject) in
+            
+            guard let responseAutocompleteSuggestions = responseObject as? [RCHAutocompleteSuggestion] else {
+                print("Result Error")
+                return
+            }
+            
+            self.autocompleteSuggestions = responseAutocompleteSuggestions.map({$0.text!})
+        }) { (responseObject, error) in
+            print(error)
+        }
     }
     
-    func showNoResults() {
-        searchResultsCollectionView.isHidden = true
-        searchProductsView.isHidden = false
-        searchProductsLabel.text = "No Results"
-        searchProductsImageView.image = UIImage(named: "icn-tabbar-shop.pdf")
+    // MARK: IBActions
+    
+    @IBAction func handleTapOnFooter(sender: UITapGestureRecognizer) {
+        resetSearch()
     }
     
+    // MARK: UISearchBarDelegate
+    
     func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
         searchProductsView.isHidden = true
-        autocompleteTableView.isHidden = false
     }
     
     func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
-        guard let searchTerm = searchBar.text else {
+        guard let searchText = searchBar.text else {
             print("Error: no search term entered")
             return
         }
-        searchForProducts(withTerm: searchTerm)
+        searchTerm = searchText
         resetSearch()
     }
     
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        // TODO: Autocomplete and live searching here
-        searchTerm = searchText
-        searchForProducts(withTerm: searchText)
-        autocompleteTableView.reloadData()
-        print(searchText)
+        if searchBar.text == "" {
+            products.removeAll()
+            autocompleteSuggestions.removeAll()
+        } else {
+            searchTerm = searchText
+            autocomplete(withQuery: searchText)
+        }
```

Comment:
I tried this before but if I move autocomplete into searchTerm.didSet then it gets called when the user selects an autocomplete suggestion as the searchTerm. Which means the autocomplete table pops up over the search results, and we don't want that here.